### PR TITLE
[docs] Fixed command not installing dependencies in development environment

### LINKED ISSUE #114: [docs] Instructions for development
Instructions for development are redundant. Similar to  https://github.com/openwisp/openwisp-users/issues/113#issuecomment-625443317 

In https://github.com/openwisp/openwisp-utils#installing-for-development
Following command does not install dependencies of [qa] and [rest]  which are required for running tests.  
```
python setup.py develop
```

### PR DESCRIPTION
**Pull request by KapilBansal320**
Replaced `python setup.py develop` with `pip install -e .[qa,rest]` for developmet version.

Closes #114

### CODE REVIEW

**Review by pandafy — CHANGES_REQUESTED** [review]
I don't think we need re-ordering of development instructions. The repetition was taken care of in https://github.com/openwisp/openwisp-utils/commit/0ba09ccb966896184f511d7f1c01ddf05467e04f. 

We still need to change `python setup.py develop` to `pip install -e .[qa,rest]` :+1:

@KapilBansal320 please revert changes for re-ordering development instructions.

**Comment by KapilBansal320** [comment]
I had reverted the changes for reordering of development instructions.

**Review by atb00ker — APPROVED** [review]
LGTM, thanks! :smile:

Merge deffered to @nemesisdesign

**Review by pandafy — APPROVED** [review]
**TLDR;** I think these changes suffice what I am expecting.

@nemesisdesign when I build the module using `python setup.py develop` (following present instructions) I get `ModuleNotFound` errors for dependencies listed in optional packages since they are used in the project (e.g. DRF).

Here is a traceback while running tests after installing with `python setup.py develop` and `pip install -r requirements-test.txt`. 
```
$ ./runtests.py 
Traceback (most recent call last):
  File "./runtests.py", line 16, in <module>
    execute_from_command_line(args)
  File "/home/openwisp/openwisp/openwisp-utils/venv2/lib/python3.7/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/home/openwisp/openwisp/openwisp-utils/venv2/lib/python3.7/site-packages/django/core/management/__init__.py", line 377, in execute
    django.setup()
  File "/home/openwisp/openwisp/openwisp-utils/venv2/lib/python3.7/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/openwisp/openwisp/openwisp-utils/venv2/lib/python3.7/site-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
  File "/home/openwisp/openwisp/openwisp-utils/venv2/lib/python3.7/site-packages/django/apps/config.py", line 90, in create
    module = import_module(entry)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'rest_framework'
```

While `pip install -e .[qa, rest]` also installs the optional packages and hence their dependencies. Therefore, no error is raised.

My main concern was when a new contributor decides to work on this repository, they will have to install those dependencies either one by one or look for other means to get this resolved which hurts developer experience.
